frontend/model/settings/vuexModule.js

```diff
@@ -70,6 +70,12 @@ const mutations = {
     state.increasedContrast = isChecked
   },
   setNotificationEnabled (state, enabled) {
+    if (state.notificationEnabled !== enabled) {
+      sbp('service-worker/setup-push-subscription').catch(e => {
+        state.notificationEnabled = false
```

Comment:
My understanding of Vuex is that changes to its state must be made synchronously, so this line probably can't be done like this...

Instead it should be set with another call to `commit` - but be careful not to end up in an infinite loop...

Reply:
You may be right, but AFAIK this works, as Vuex has observers set up. For example, if you do `sbp('state/vuex/state').notifications.notificationEnabled = false` and observe the switcher, it works, even if the change happened 'asynchronously'.

To clarify what I mean, the _setter_ need to be synchronous. That's obvious. If you call `state/vuex/commit` and it were async it'd be weird and lead to unintended side effects. However, this code is similar to having some code like:

```js
sbp('okTurtles.events/on', PUSH_SUB_FAILED, () => {
  sbp('state/vuex/state').notifications.notificationEnabled = false
})
```

Do you have documentation on how this is supposed to work?

Reply:
That would possibly work but would be a DRY violation

Reply:
@corrideat Hmm, fair point on the dry violation.

Then in that case, maybe using `sbp('state/vuex/commit', 'setNotificationEnabled', false)` inside of the `.catch` would work?

The `if` guard should prevent an infinite loop...